semanticcpg/src/main/scala/io/shiftleft/semanticcpg/dotgenerator/CallGraphGenerator.scala

```diff
@@ -0,0 +1,36 @@
+package io.shiftleft.semanticcpg.dotgenerator
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{Method, StoredNode}
+import io.shiftleft.semanticcpg.dotgenerator.DotSerializer.{Edge, Graph}
+import io.shiftleft.semanticcpg.language._
+
+import scala.collection.mutable
+
+class CallGraphGenerator {
+
+  def generate(cpg: Cpg): Graph = {
+    val subgraph = mutable.HashMap.empty[String, Seq[StoredNode]]
+    val vertices = cpg.all.collect { case m: Method => m }.l
+    val edges = vertices.flatMap { srcMethod =>
+      storeInSubgraph(srcMethod, subgraph)
+      srcMethod.call.flatMap { child =>
+        child.callOut.map { tgt =>
+          storeInSubgraph(tgt, subgraph)
+          Edge(srcMethod, tgt, label = child.dispatchType.stripSuffix("_DISPATCH"))
+        }
+      }
+    }.distinct
```

Comment:
```suggestion
    val edges = for {
      srcMethod <- vertices
      _ = storeInSubgraph(srcMethod, subgraph)
      child <- srcMethod.call
      tgt <- child.callOut
    } yield {
      storeInSubgraph(tgt, subgraph)
      Edge(srcMethod, tgt, label = child.dispatchType.stripSuffix("_DISPATCH"))
    }.distinct
```

The for comprehension saves the reader from parsing the flatMap/map etc... a matter of style, up to you really.
Note that I didn't ask the compiler for feedback on the above yet...